### DISCUSSION TITLE
Date time formats for specific / rare language

### DISCUSSION THREAD

**Discussion by RollingTL** [intlify/vue-i18n#689] [Q&A]
Of three languages - English, Russian, Kazakh, only two show correct date.
English and Russian dates are displayed correctly, but Kazakh dates are shown in English.

How can I add Kazakh dates support? 
Kazakh locale short name is '**kz**'

And I wonder where is all date specific translations are stored? How to manage them?

Thanks

I use Vuejs.3 plus TypeScript
Inside Vue templates I use following code:
`$d(my_date, 'short') `

**Main.ts** 
```
import { createI18n } from 'vue-i18n'
import messageDictionary from '@/i18n/messageDictionary'
import dateTimeFormats from '@/i18n/dateTimeFormats'
const i18n = createI18n({
  globalInjection: true,
  legacy: false,
  locale: 'ru',
  fallbackLocale: 'ru',
  availableLocales: ['ru', 'kz', 'en'],
  messages: messageDictionary(),
  datetimeFormats: dateTimeFormats()
})
const app = createApp(App)
app.use(i18n)
app.use(router).mount('#app')
```

**dateTimeFormats.ts** 
```
export default () => ({
  en: {
    short: {
      year: 'numeric',
      month: 'short',
      day: 'numeric'
    },
    long: {
      year: 'numeric',
      month: 'long',
      day: 'numeric',
      weekday: 'short',
      hour: 'numeric',
      minute: 'numeric',
      hour12: false
    }
  },
  ru: {
    short: {
      year: 'numeric',
      month: 'short',
      day: 'numeric'
    },
    long: {
      year: 'numeric',
      month: 'long',
      day: 'numeric',
      weekday: 'short',
      hour: 'numeric',
      minute: 'numeric',
      hour12: false
    }
  },
  kz: {
    short: {
      year: 'numeric',
      month: 'short',
      day: 'numeric'
    },
    long: {
      year: 'numeric',
      month: 'long',
      day: 'numeric',
      weekday: 'short',
      hour: 'numeric',
      minute: 'numeric',
      hour12: false
    }
  }
})
```

**Comment by kazupon** [comment]
DateTime information needs to be specified in the datetimeFormats option in vue-i18n.
https://vue-i18n.intlify.dev/guide/essentials/datetime.html

DatetimeFormat option, which has the same options as Intl.DateTimeFormat.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat

DatetimeFormat. After specifying that option, you can use `$d` (Composition 
API, d is exported from useI18n) to localize it.

**Reply by parallels999** [reply]
@RollingTL is this **answered**?

**Comment by RollingTL** [comment]
I think that date and time info is taken from browser / system.  If I install for example Kazakh language on my machine then "$d(my_date, 'short')" will work in Kazakh.  Will check and report here.
